stable_plugins/visualization/file_types/qasm_visualization.py

```diff
@@ -0,0 +1,252 @@
+                        required=True,
+                    )
+                ],
+                data_output=[
+                    DataMetadata(
+                        data_type="*",
+                        content_type=["image/svg+xml"],
+                        required=True,
+                    )
+                ],
+            ),
+            tags=[],
+        )
+
+
+@QASM_BLP.route("/ui/")
+class MicroFrontend(MethodView):
+    """Micro frontend for the QASM visualization plugin."""
+
+    @QASM_BLP.html_response(
+        HTTPStatus.OK, description="Micro frontend of the QASM visualization plugin."
+    )
+    @QASM_BLP.arguments(
+        QasmInputParametersSchema(
+            partial=True, unknown=EXCLUDE, validate_errors_as_result=True
+        ),
+        location="query",
+        required=False,
+    )
+    @QASM_BLP.require_jwt("jwt", optional=True)
+    def get(self, errors):
+        """Return the micro frontend."""
+        return self.render(request.args, errors, False)
+
+    @QASM_BLP.html_response(
+        HTTPStatus.OK, description="Micro frontend of the QASM visualization plugin."
+    )
+    @QASM_BLP.arguments(
+        QasmInputParametersSchema(
+            partial=True, unknown=EXCLUDE, validate_errors_as_result=True
+        ),
+        location="form",
+        required=False,
+    )
+    @QASM_BLP.require_jwt("jwt", optional=True)
+    def post(self, errors):
+        """Return the micro frontend with prerendered inputs."""
+        return self.render(request.form, errors, not errors)
+
+    def render(self, data: Mapping, errors: dict, valid: bool):
+        plugin = QasmVisualization.instance
+        if plugin is None:
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR)
+        schema = QasmInputParametersSchema()
+        return Response(
+            render_template(
+                "qasm_visualization.html",
+                name=plugin.name,
+                version=plugin.version,
+                schema=schema,
+                valid=valid,
+                values=data,
+                errors=errors,
+                example_values=url_for(f"{QASM_BLP.name}.MicroFrontend"),
+            )
+        )
+
+
+@QASM_BLP.route("/circuits/")
+def get_circuit_image():
+    circuit_url = request.args.get("circuit-url", None)
+    if circuit_url is None:
+        abort(HTTPStatus.NOT_FOUND)
+    filename = hashlib.md5(circuit_url.encode("utf-8")).hexdigest() + ".png"
+    path = pathlib.Path(__file__).parent.absolute() / "img" / filename
+    if os.path.exists(path):
+        return send_file(path, mimetype="image/png")
+    try:
+        with open_url(circuit_url) as quasm_response:
+            circuit_qasm = quasm_response.text
+    except HTTPError:
+        TASK_LOGGER.info(f"Could not load circuit from {circuit_url}")
+        abort(HTTPStatus.NOT_FOUND)
+    circuit = QuantumCircuit.from_qasm_str(circuit_qasm)
+    fig = circuit.draw(output="mpl")
+    fig.savefig(path)
```

Comment:
You can do this inside a task and just await the result with a timeout of <10s in this response. In case of a timeout error respond wit the appropriate error code (and maybe the browser will try again later to see if the image is available automatically).